cumulus/polkadot-parachain/polkadot-parachain-lib/src/nodes/manual_seal.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::common::{
+	rpc::BuildRpcExtensions as BuildRpcExtensionsT,
+	spec::{BaseNodeSpec, BuildImportQueue, NodeSpec as NodeSpecT},
+	types::{Hash, ParachainBlockImport, ParachainClient},
+	ConstructNodeRuntimeApi, NodeBlock,
+};
+use codec::Encode;
+use cumulus_client_parachain_inherent::{MockValidationDataInherentDataProvider, MockXcmConfig};
+use cumulus_primitives_core::ParaId;
+use sc_consensus::{DefaultImportQueue, LongestChain};
+use sc_consensus_manual_seal::rpc::{ManualSeal, ManualSealApiServer};
+use sc_network::NetworkBackend;
+use sc_service::{build_polkadot_syncing_strategy, Configuration, PartialComponents, TaskManager};
+use sc_telemetry::TelemetryHandle;
+use sp_runtime::traits::Header;
+use sp_timestamp::Timestamp;
+use std::{marker::PhantomData, sync::Arc};
+
+/// Build the import queue for a manual seal node.
+pub struct BuildManualSealImportQueue<Block, RuntimeApi>(PhantomData<(Block, RuntimeApi)>);
+
+impl<
+		Block: NodeBlock,
+		RuntimeApi: ConstructNodeRuntimeApi<Block, ParachainClient<Block, RuntimeApi>>,
+	> BuildImportQueue<Block, RuntimeApi> for BuildManualSealImportQueue<Block, RuntimeApi>
+{
+	fn build_import_queue(
+		client: Arc<ParachainClient<Block, RuntimeApi>>,
+		_block_import: ParachainBlockImport<Block, RuntimeApi>,
+		config: &Configuration,
+		_telemetry_handle: Option<TelemetryHandle>,
+		task_manager: &TaskManager,
+	) -> sc_service::error::Result<DefaultImportQueue<Block>> {
+		Ok(sc_consensus_manual_seal::import_queue(
+			Box::new(client.clone()),
+			&task_manager.spawn_essential_handle(),
+			config.prometheus_registry(),
+		))
+	}
+}
+
+pub struct ManualSealNode<NodeSpec>(PhantomData<NodeSpec>);
+
+impl<NodeSpec: NodeSpecT> BaseNodeSpec for ManualSealNode<NodeSpec> {
+	type Block = NodeSpec::Block;
+	type RuntimeApi = NodeSpec::RuntimeApi;
+	type BuildImportQueue = BuildManualSealImportQueue<Self::Block, Self::RuntimeApi>;
```

Comment:
Nit: We could implement BuildImportQueue directly for `ManualSealNode` instead of having the extra struct. Just a suggestion, do as you like.

Reply:
Good point ! Done.